app/eventPage/[id].tsx

```diff
@@ -1,18 +1,27 @@
-import { useLocalSearchParams } from "expo-router";
-import React from "react";
+import { Link, useLocalSearchParams } from "expo-router";
+import React, { useEffect, useState } from "react";
 import { ScrollView, SizableText, Text, View, XStack, YStack } from "tamagui";
 
 import { FloatingActionButton } from "@/components/FloatingActionButton";
 import { IconSymbol } from "@/components/ui/IconSymbol";
 import { useEventById } from "@/hooks/use-event";
+import { useStorage } from "@/hooks/use-storage";
 import { useUser } from "@/hooks/use-user";
 
 export default function EventPage() {
   const { id } = useLocalSearchParams();
   const { data: events } = useEventById(Number(id));
   const { data: user } = useUser();
   const event = events?.docs[0];
-  // const navigation = useNavigation();
+
+  const storage = useStorage();
+  const [joinStatus, setJoinStatus] = useState<string | null>(null);
+
+  useEffect(() => {
+    if (event) {
+      storage.getString(event.eventCode).then(setJoinStatus);
+    }
+  }, [event, storage]);
```

Comment:
This works, but ideally I would see react query be used here so that we have all the different statuses and cache in place